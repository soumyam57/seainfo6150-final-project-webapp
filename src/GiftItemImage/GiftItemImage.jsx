import React from "react";
import PropTypes from "prop-types";
import styles from "./GiftItemImage.module.css";
import GiftItemDetail from "../GiftItemDetail/GiftItemDetail";
import { Link } from "react-router-dom";

const GiftItemImage = (props) => {
  return (
    <div className={styles.flex}>
      <p className={styles.text}>{props.title}</p> 
      <img className={styles.image} 
          src={props.url} alt="">
      </img>
      <p className={styles.text}>Price: {props.price}</p> 
    </div>
    
  );
};

GiftItemImage.propTypes = {
  url: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};
export default GiftItemImage;