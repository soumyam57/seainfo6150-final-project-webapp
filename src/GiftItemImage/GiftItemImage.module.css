
.image{
    padding: 1%;
    width: 70%;
    border-radius: 20%;
    box-shadow:0px 0px 10px 5px grey;
    border: 5px solid maroon;
    
}

.text {
    padding:0px 10px 10px 0px;
    color: gainsboro;
    font-weight: bolder;
    text-align: start;
  }

.flex{
    display: flex;
    gap: 1%;
    flex-direction: column;
    align-items: center;
}
