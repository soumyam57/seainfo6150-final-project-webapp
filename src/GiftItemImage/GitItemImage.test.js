import React from "react";
import { render } from "@testing-library/react";
import GiftItemImage from "./GiftItemImage.jsx";

it("renders the BooksListItemImage component correctly", () => {
  const { container } = render(
    <GiftItemImage
      url="https://images-na.ssl-images-amazon.com/images/I/61aaipDiGcL._AC_SL1200_.jpg"
    />
  );
  expect(container).toMatchSnapshot();
});