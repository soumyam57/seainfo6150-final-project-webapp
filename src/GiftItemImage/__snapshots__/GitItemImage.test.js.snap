// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BooksListItemImage component correctly 1`] = `
<div>
  <div>
    <img
      alt=""
      class="image"
      src="https://images-na.ssl-images-amazon.com/images/I/61aaipDiGcL._AC_SL1200_.jpg"
    />
  </div>
</div>
`;
